### PR TITLE
Ecto: querying basics

### PR DESCRIPTION
**Pull request by SophieDeBenedetto**
I'm thinking this lesson should be geared towards "querying basics" and then we can do a separate lesson on writing more complex queries, taking a closer look at joins etc. I definitely hesitated on how much detail to add here so would love some feedback on that in particular.

### CODE REVIEW

**Comment by SophieDeBenedetto** [comment]
Hey @notactuallypagemcconnell and @gj! What do you think about moving forward with a shorter querying lesson here, something like "Querying Basics", while you guys continue to work on a more in-depth one? Would love to get something out soon and I have some time over the holidays so I'd be happy to put something simple together.

**Comment by doomspork** [comment]
Sounds like a good idea to me. Happy to help with some sections if need be @SophieDeBenedetto 😁

**Comment by SophieDeBenedetto** [comment]
@doomspork just pushed up a first draft. I'm thinking this lesson should be geared towards "querying basics" and then we can do a separate lesson on writing more complex queries, taking a closer look at `joins` etc. I definitely hesitated on how much detail to add here. Let me know what you think.

File: _data/contents.yml [hunk 1]
```diff
@@ -42,6 +42,8 @@ specifics:
   - ets
   - mnesia
   - debugging
+ecto:
```
Comment (doomspork) [inline _data/contents.yml]: If you rebase on master there should be other lessons here.

File: en/lessons/ecto/querying.md [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+
+We _can't_ access those associated characters unless we preload them. There are a few different way to preload records with Ecto.
+
+#### Preloading With Two Queries
+
+The following query will preload associated records in a _separate_ query.
+
+```elixir
+iex> import Ecto.Query
+Ecto.Query
```
Comment (doomspork) [inline en/lessons/ecto/querying.md]: We can omit output like this for brevity 👍

File: en/lessons/ecto/querying.md [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+    tagline: "Something about video games",
+    title: "Ready Player One"
+  }
+]
+```
+
+This allows us to execute just one database call. It also has the added benefit of allowing us to select and filter both movies and associated actors in the same query. For example, this approach allows us to query for all movies where the associated actors meet certain conditions using a `join` statement. Something like:
+
+```elixir
+Repo.all from m in Movie,
```
Comment (doomspork) [inline en/lessons/ecto/querying.md]: Is there a better way to express this?  We should have `()` at the least.

File: en/lessons/ecto/querying.md [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+In the example above, we are joining on an Ecto schema, `m in Movie`. We can also join on an Ecto query. Let's say our movies table has a column `stars`, where we store the "star rating" of the film, a number 1-5.
+
+```elixir
+movies = from m in Movie, where: [stars: 5]
+from c in Character,
+  join: ^movies,
+  on: [id: c.movie_id], # keyword list
+  where: c.name == "Video Game Guy",
+  select: {m.title, c.name}
+```
```
Comment (doomspork) [inline en/lessons/ecto/querying.md]: We should wrap up the lesson with some comment 👍

**Comment by SophieDeBenedetto** [comment]
@doomspork thanks for your feedback! Addressed your comments 🚀

File: en/lessons/ecto/querying.md [hunk 5]
```diff
@@ -21,9 +21,8 @@ We can use the `Repo.get/3` function to fetch a record from the database given i
 Let's take a look at an example. Below, we'll get the movie with and ID of 1:
 
 ```elixir
-iex> alias Example.Repo
-iex> alias Example.Movie
-iex> Repo.get(Movie, 1)
+iex> alias Example.{Repo, Movie}
+©iex> Repo.get(Movie, 1)
```
Comment (doomspork) [inline en/lessons/ecto/querying.md]: There's a copyright symbol in here now 👍

**Comment by SophieDeBenedetto** [comment]
@doomspork fixed that copyright symbol :) 
This lesson should go out either with or after the associations lesson https://github.com/elixirschool/elixirschool/pull/1553 since it builds on the same domain model. That PR still needs approval so take a look if you have a chance

**Pull request merged**